Jetcaster/app/src/main/java/com/example/jetcaster/util/WindowInfoUtil.kt

```diff
@@ -41,6 +48,15 @@ fun isTableTopPosture(foldFeature: FoldingFeature?): Boolean {
         foldFeature.orientation == FoldingFeature.Orientation.HORIZONTAL
 }
 
-fun isBookPosture(foldFeature: FoldingFeature?) =
-    foldFeature?.state == FoldingFeature.State.HALF_OPENED &&
+@OptIn(ExperimentalContracts::class)
+fun isBookPosture(foldFeature: FoldingFeature?): Boolean {
+    contract { returns(true) implies (foldFeature != null) }
+    return foldFeature?.state == FoldingFeature.State.HALF_OPENED &&
         foldFeature.orientation == FoldingFeature.Orientation.VERTICAL
+}
+
+@OptIn(ExperimentalContracts::class)
+fun isSeparating(foldFeature: FoldingFeature?): Boolean {
+    contract { returns(true) implies (foldFeature != null) }
+    return foldFeature?.state == FoldingFeature.State.FLAT && foldFeature.isSeparating
```

Comment:
Nothing to do with this PR but `isSeparating` isn't the best of the names 😅  I don't know what it means!

Reply:
Ack - renamed to `isSeparatingPosture`

The idea is that this returns `true` when the window is separated into multiple physical areas. See [`FoldingFeature#isSeparating`](https://developer.android.com/reference/androidx/window/layout/FoldingFeature#isSeparating()) for more information.